code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -229,8 +180,102 @@ Contains:
 		src.part3 = null
 		src.c_state(src.part1.timing)
 		boutput(usr, "<span class='notice'>You remove the timer/igniter assembly from the beaker.</span>")
+		//since the assembly is free of beakers, let's add the assembly components again
+		// Timer-Igniter Assembly + wired pipebomb -> timer-igniter pipebomb
+		src.AddComponent(/datum/component/assembly, /obj/item/pipebomb/frame, PROC_REF(pipebomb_frame_assembly), TRUE)
+		// Timer-Igniter Assembly + pipebomb -> timer-igniter pipebomb
+		src.AddComponent(/datum/component/assembly, /obj/item/pipebomb/bomb, PROC_REF(pipebomb_assembly), TRUE)
+		// Timer-Igniter Assembly + beaker -> timer-igniter beakerbomb
+		src.AddComponent(/datum/component/assembly, /obj/item/reagent_containers/glass/beaker, PROC_REF(beakerbomb_assembly), TRUE)
 	else boutput(usr, "<span class='alert'>That doesn't have a beaker attached to it!</span>")
 
+// Timer-igniter-assemblies
+
+///Beakerbomb-assembly
+/obj/item/assembly/time_ignite/proc/beakerbomb_assembly(var/atom/to_combine_atom, var/mob/user)
+	if(src.status)
+		boutput(user, "You need to unsecure the timer first.")
+		return
+	var/obj/item/reagent_containers/glass/beaker/manipulated_beaker = to_combine_atom
+	src.part3 = manipulated_beaker
+	manipulated_beaker.master = src
+	manipulated_beaker.layer = initial(manipulated_beaker.layer)
+	user.u_equip(manipulated_beaker)
+	manipulated_beaker.set_loc(src)
+	src.c_state(0)
+	boutput(user, "You attach the timer/igniter assembly to the beaker.")
+
+	//Since we completed the assembly, remove all assembly components
+	var/list/assembly_component_list = src.GetComponents(/datum/component/assembly)
+	for (var/datum/component/component_to_remove in assembly_component_list)
+		component_to_remove.RemoveComponent()
```

Comment:
Feel like there should be some more elegant way of handling this than this, considering how it seems to be being universally applied 🤷